nilearn/tests/test_masking.py

```diff
@@ -1,371 +1,483 @@
+    "Expected dimension is 3D and you provided "
+    "a %s image"
+)
+AFFINE_EYE = np.eye(4)
+SHAPE_2D = (9, 9)
+SHAPE_3D = (9, 9, 9)
+SHAPE_4D = (9, 9, 9, 9)
+
+
+def _confounds_regression(standardize_signal=True, standardize_confounds=True):
+    img, mask, conf = _simu_img()
+    masker = NiftiMasker(
+        standardize=standardize_signal,
+        standardize_confounds=standardize_confounds,
+        detrend=False,
+        mask_img=mask,
+    ).fit()
+    tseries = masker.transform(img, confounds=conf)
+    if standardize_confounds:
+        conf = StandardScaler(with_std=False).fit_transform(conf)
+    cov_mat = _cov_conf(tseries, conf)
+    return np.sum(np.abs(cov_mat))
 
-_TEST_DIM_ERROR_MSG = ("Input data has incompatible dimensionality: "
-                       "Expected dimension is 3D and you provided "
-                       "a %s image")
 
 def _simu_img():
     # Random confounds
     conf = 2 + np.random.randn(100, 6)
     # Random 4D volume
     vol = 100 + 10 * np.random.randn(5, 5, 2, 100)
-    img = Nifti1Image(vol, np.eye(4))
+    img = Nifti1Image(vol, AFFINE_EYE)
     # Create an nifti image with the data, and corresponding mask
-    mask = Nifti1Image(np.ones([5, 5, 2]), np.eye(4))
+    mask = Nifti1Image(np.ones([5, 5, 2]), AFFINE_EYE)
     return img, mask, conf
 
+
 def _cov_conf(tseries, conf):
     conf_n = StandardScaler().fit_transform(conf)
-    tseries_n = StandardScaler().fit_transform(tseries)
+    StandardScaler().fit_transform(tseries)
     cov_mat = np.dot(tseries.T, conf_n)
     return cov_mat
 
-def _confounds_regression(standardize_signal=True, standardize_confounds=True):
-    rng = np.random.RandomState(42)
-    img, mask, conf = _simu_img()
-    masker = NiftiMasker(standardize=standardize_signal,
-                         standardize_confounds=standardize_confounds,
-                         detrend=False,
-                         mask_img=mask).fit()
-    tseries = masker.transform(img, confounds=conf)
-    if standardize_confounds:
-        conf = StandardScaler(with_std=False).fit_transform(conf)
-    cov_mat = _cov_conf(tseries, conf)
-    return np.sum(np.abs(cov_mat))
-
-def test_high_variance_confounds():
-    rng = np.random.RandomState(42)
-    img, mask, conf = _simu_img()
-    hv_confounds = high_variance_confounds(img)
-    masker1 = NiftiMasker(standardize=True, detrend=False,
-                          high_variance_confounds=False,
-                          mask_img=mask).fit()
-    tseries1 = masker1.transform(img, confounds=[hv_confounds, conf])
-    masker2 = NiftiMasker(standardize=True, detrend=False,
-                          high_variance_confounds=True,
-                          mask_img=mask).fit()
-    tseries2 = masker2.transform(img, confounds=conf)
-    np.testing.assert_array_equal(tseries1, tseries2)
 
+@pytest.mark.parametrize(
+    "standardize_signal, expected",
+    [(False, 10.0), ("zscore", 1.0), (True, 1.0), ("psc", 10.0)],
```

Comment:
I think this should now be:

```suggestion
    [(False, 10.0), ("zscore_sample", 1.0), (True, 1.0), ("psc", 10.0)],
```

See https://github.com/nilearn/nilearn/pull/3474